### DISCUSSION TITLE
The selection behavior of the list view in MVUX is strange.

### DISCUSSION THREAD

**Discussion by dimohy** [unoplatform/uno#15654] [Q&A]
I've only ever used MVVM, but I'm intrigued by Uno's MVUX, so I did a quick test.
For lists, it doesn't work as expected.

Normal behavior in Windows projects, issue in Android and WebAssembly.
When selecting a list one by one from the bottom to the top, it is selected, but when selecting the list from top to bottom, it is not selected.

> Checked on Visual Studio 2022 17.9.1, .NET 8, Uno Platform 5.1.75

![ezgif-4-15d37d71fc](https://github.com/unoplatform/uno/assets/58620778/24019578-dca3-4223-8a83-94a1695792fe)

| MainModel.cs
```csharp
    private readonly List<Entity> _entities = [
        new ("Entity 1"),
        new ("Entity 2"),
        new ("Entity 3"),
        new ("Entity 4"),
        new ("Entity 5"),
        new ("Entity 6"),
        new ("Entity 7"),
        new ("Entity 8"),
        new ("Entity 9"),
        new ("Entity 10"),
    ];

    public IListState<Entity> Entities => ListState<Entity>.Value(this, () => _entities.ToImmutableList())
        .Selection(SelectedEntity);

    public IState<Entity> SelectedEntity => State<Entity>.Empty(this);
```

| MainPage.xaml
```xaml
...
      <Border Background="Red">
        <TextBlock Text="{Binding SelectedEntity.Name}" />
      </Border>

      <ListView ItemsSource="{Binding Entities}">
        <ListView.ItemTemplate>
          <DataTemplate>
            <TextBlock Text="{Binding Name}" />
          </DataTemplate>
        </ListView.ItemTemplate>
      </ListView>
...
```

Does anyone know why this is the case?

**Comment by CalebSerafin** [comment]
I've experienced this issue as well with Lists and ComboBoxes. Since I'm not the only person I'll guess that this might be a bug that could be related to https://github.com/unoplatform/uno/issues/14792 and https://github.com/unoplatform/uno/issues/15380

I workaround this by having a `SelectionChanged` handler in the code behind that then sets the state. The item source is provided by `IFeed<ImmutableList<ItemThing>>` (Or any other Feed with a collection) as `ListFeed<ItemThing>`s seem to cause this issue.

**Comment by dr1rrb** [comment]
Indeed : what I can see is that when you select the item right after the currently selected one, selection is not updated properly. But if you select an item that is at least 2 items after the lastly clicked (even if it does not appears as selected), then it works.

Most probably just an index issue (I also noticed that sometimes when you select an item, the item above is also selected), but definitely a bug!

**Reply by sasakrsmanovic** [reply]
@dimohy would you be kind enough to convert this to an issue please, adding any other information you have to it. Thanks so much for taking the time to document it.
